runtime/kubernetes/container.go

```diff
@@ -238,6 +238,10 @@ func (c *client) setupContainerEnvironment(ctn *pipeline.Container) error {
 func (c *client) TailContainer(ctx context.Context, ctn *pipeline.Container) (io.ReadCloser, error) {
 	c.Logger.Tracef("tailing output for container %s", ctn.ID)
 
+	// create a logsContext that will be canceled at the end of this
+	logsContext, logsDone := context.WithCancel(ctx)
+	defer logsDone()
+
```

Comment:
Aargh. This is wrong.

We can't create a context here because we expect (at least part of) logsFunc to continue running after TailContainer returns. Then StreamStep/StreamService/etc read from the returned ReadCloser until the end of the stream. If the context gets canceled at the end of TailContainer, then the streaming is stopped before Stream* has read anything from it.